### PR TITLE
xscreensaver 5.39

### PR DESCRIPTION
**Pull request by Mno-hime**
* Making it 64-bit-only
* added couple of build dependencies not found autumatically as
  `usr/lib/xscreensaver/hacks/` are likely not checked for binary `so`
  requirements
* Runtested: screen blank, screen poweroff, password login, various
  timeouts.
* https://www.illumos.org/issues/9581 still applies.

### CODE REVIEW

File: components/desktop/xscreensaver/Makefile [hunk 1]
```diff
@@ -53,6 +53,8 @@ LDFLAGS+=	-I$(@D)/mesa/GL
 
 CONFIGURE_ENV+=	"XGETTEXT=/usr/gnu/bin/xgettext"
 
+CONFIGURE_BINDIR.64=	$(CONFIGURE_BINDIR.32)
```
Comment (alarcher) [inline components/desktop/xscreensaver/Makefile]: Could you use PREFERRED_BITS=64 before include of shared-macros.mk instead?

File: components/desktop/xscreensaver/Makefile [hunk 2]
```diff
@@ -70,28 +72,34 @@ CONFIGURE_OPTIONS+=	--with-configdir=/usr/lib/xscreensaver/config/control-center
...
-build:		$(BUILD_32)
+build:		$(BUILD_64)
 
-install:	$(INSTALL_32) ../rss-glx/build/i86/.built
+install:	$(INSTALL_64) ../rss-glx/build/i86/.built
 	perl ../rss-glx/build/i86/utils/rss-glx_install.pl $(PROTO_DIR)/usr/share/X11/app-defaults/XScreenSaver
 test:		$(NO_TESTS)
 
 REQUIRED_PACKAGES += gnome/config/gconf
+REQUIRED_PACKAGES += image/library/libjpeg
```
Comment (alarcher) [inline components/desktop/xscreensaver/Makefile]: Could you make sure that libjpeg8-turbo is used? Ref: https://wiki.openindiana.org/oi/JPEG+8

File: components/desktop/xscreensaver/Makefile [hunk 3]
```diff
@@ -70,28 +72,34 @@ CONFIGURE_OPTIONS+=	--with-configdir=/usr/lib/xscreensaver/config/control-center
...
+REQUIRED_PACKAGES += image/library/libpng16
+REQUIRED_PACKAGES += library/c++/harfbuzz
 REQUIRED_PACKAGES += library/desktop/atk
 REQUIRED_PACKAGES += library/desktop/gdk-pixbuf
 REQUIRED_PACKAGES += library/desktop/gtk2
 REQUIRED_PACKAGES += library/desktop/pango
 REQUIRED_PACKAGES += library/glib2
 REQUIRED_PACKAGES += library/libxml2
 REQUIRED_PACKAGES += library/perl-5/xml-parser
-REQUIRED_PACKAGES += runtime/perl-522
```
Comment (alarcher) [inline components/desktop/xscreensaver/Makefile]: Alexander suggested that both Perl versions should be kept. I am not sure it sill applies.

**Review by alarcher — CHANGES_REQUESTED** [review]

File: components/desktop/xscreensaver/Makefile (hunk 3 above)
Comment (Mno-hime) [reply components/desktop/xscreensaver/Makefile]: OK, I kept Perl 5.22.

**Review by alarcher — APPROVED** [review]

**Pull request merged**